pkg/cloudprovider/aws.go

```diff
@@ -0,0 +1,39 @@
+package cloudprovider
+
+import (
+	"fmt"
+	"strings"
+
+	v1 "k8s.io/api/core/v1"
+)
+
+const (
+	// AWSProviderName AWS cloud provider name
+	AWSProviderName = "aws"
+)
+
+type aws struct {
+	isEks bool
+}
+
+func (a *aws) init(nodeList *v1.NodeList) error {
+	for _, node := range nodeList.Items {
```

Comment:
This seems very similar to the loop at line `https://github.com/libopenstorage/operator/pull/789/files#diff-a6af1de71edb43b701d83b9ee10e345cf3c6ef3fc738ee3132c3309b6958896dR65`... any reason we need it in both places and/or can we reduce code duplication somehow?

Reply:
at cloud provider init we try this to check the provider name. given different providers may use different way to init, here just aws provider we need to scan nodes again not necessary other providers. to make it clean and isolate the general init and specific provider init I put another loop here. don't have a better idea yet, but could be improved once we have all provider implementations in the future.